### PR TITLE
Replace TreeSitter.Range and TreeSitter.Span with `semantic-source`

### LINKED ISSUE #213: Replace Range/Span with semantic-source dependency
We want to share a common representation of ranges and spans between various packages involved in `semantic`; to that end, we extracted `semantic`’s `Range`, `Span`, `Location`, &c. types into a new package, `semantic-source`.

`semantic-source` doesn’t depend on `tree-sitter` so that using ranges/spans doesn’t require a package to be tied to any particular parser. But we still want to be able to `Unmarshal` directly to these types so as to avoid orphan instances, unnecessary copying, or inconvenience. Therefore, we should replace `TreeSitter.Range` & `TreeSitter.Span` with a dependency on `semantic-source`, and derive `Unmarshal` instances for `semantic-source`’s `Range`, `Span`, and `Location` types.

### PR DESCRIPTION
**Pull request by aymannadeem**
Fixes #213

### CODE REVIEW

File: tree-sitter-python/tree-sitter-python.cabal [hunk 1]
```diff
@@ -23,6 +23,7 @@ library
                      , filepath         ^>= 1.4
                      , template-haskell  >= 2.12.0.0 && < 2.15.0.0
                      , tree-sitter-python-internal
+                     , semantic-source   >= 0.0.0.0
```
Comment (patrickt) [inline tree-sitter-python/tree-sitter-python.cabal]: You probably want ^>= and not >= (since that would match all future versions of semantic-source).

Comment (aymannadeem) [reply tree-sitter-python/tree-sitter-python.cabal]: Ohhh I got it the other way around. Thx!

File: tree-sitter/tree-sitter.cabal [hunk 2]
```diff
@@ -36,6 +36,7 @@ library
                      , text                 ^>= 1.2.3.1
                      , unordered-containers ^>= 0.2.9
                      , containers            >= 0.6.0.1
+                     , semantic-source       >= 0.0.0.0
```
Comment (patrickt) [inline tree-sitter/tree-sitter.cabal]: Here, too.

Comment (patrickt) [reply tree-sitter/tree-sitter.cabal]: But you’re good to go after that.

**Pull request merged**